From e648875cd1d004dfaea5fd21ff45eae2a01d0d86 Mon Sep 17 00:00:00 2001
From: Markus Niebel <Markus.Niebel@tq-group.com>
Date: Wed, 15 Jan 2020 18:31:31 +0100
Subject: [PATCH 4/8] tqma6: increase of kernel size for spi nor

Signed-off-by: Markus Niebel <Markus.Niebel@tq-group.com>
---
 include/configs/tqma6.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/configs/tqma6.h b/include/configs/tqma6.h
index bf43c2e2a3..78a161a756 100644
--- a/include/configs/tqma6.h
+++ b/include/configs/tqma6.h
@@ -180,8 +180,8 @@
 #define TQMA6_FDT_SECTOR_START		0x0a /* 8 Sector u-boot, 2 Sector env */
 #define TQMA6_FDT_SECTOR_COUNT		0x01
 
-#define TQMA6_KERNEL_SECTOR_START	0x10
-#define TQMA6_KERNEL_SECTOR_COUNT	0x60
+#define TQMA6_KERNEL_SECTOR_START	0x10 /* offset 1 MB */
+#define TQMA6_KERNEL_SECTOR_COUNT	0xa0 /* 10 MB */
 
 #define TQMA6_EXTRA_BOOTDEV_ENV_SETTINGS                                       \
 	"mmcblkdev=0\0"                                                        \
-- 
2.17.1

