From 8eeb2088975a1fb0346e9117e23c10782c7fccfb Mon Sep 17 00:00:00 2001
From: Markus Niebel <Markus.Niebel@tq-group.com>
Date: Fri, 11 Jan 2019 15:12:10 +0100
Subject: [PATCH 04/24] board: tqma8mq-mba8mx: include order in spl.c

Signed-off-by: Markus Niebel <Markus.Niebel@tq-group.com>
---
 board/tqc/tqma8mx/spl.c | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/board/tqc/tqma8mx/spl.c b/board/tqc/tqma8mx/spl.c
index a8dd34ac2a..94392e77a2 100644
--- a/board/tqc/tqma8mx/spl.c
+++ b/board/tqc/tqma8mx/spl.c
@@ -5,22 +5,22 @@
  */
 
 #include <common.h>
-#include <spl.h>
-#include <asm/io.h>
-#include <errno.h>
-#include <asm/io.h>
-#include <asm/imx-common/iomux-v3.h>
+#include <asm/arch/clock.h>
 #include <asm/arch/ddr_memory_map.h>
 #include <asm/arch/imx8mq_pins.h>
 #include <asm/arch/sys_proto.h>
-#include <power/pmic.h>
-#include <power/pfuze100_pmic.h>
-/* #include "../common/pfuze.h" */
-#include <asm/arch/clock.h>
 #include <asm/imx-common/gpio.h>
+#include <asm/imx-common/iomux-v3.h>
 #include <asm/imx-common/mxc_i2c.h>
+#include <asm/io.h>
+#include <errno.h>
 #include <fsl_esdhc.h>
 #include <mmc.h>
+#include <power/pmic.h>
+#include <power/pfuze100_pmic.h>
+/* #include "../common/pfuze.h" */
+#include <spl.h>
+
 #include "ddr/ddr.h"
 
 DECLARE_GLOBAL_DATA_PTR;
-- 
2.17.1

