From 16bb58c6b52770127b7b3ef08bac593c0f4c1e17 Mon Sep 17 00:00:00 2001
From: Markus Niebel <Markus.Niebel@tq-group.com>
Date: Mon, 12 Nov 2018 16:12:34 +0100
Subject: [PATCH] tqma8qx: fix header guard in board config header

Signed-off-by: Markus Niebel <Markus.Niebel@tq-group.com>
---
 include/configs/tqma8qx.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/configs/tqma8qx.h b/include/configs/tqma8qx.h
index 16fe2edfc4..f2e6253994 100644
--- a/include/configs/tqma8qx.h
+++ b/include/configs/tqma8qx.h
@@ -4,8 +4,8 @@
  * SPDX-License-Identifier:	GPL-2.0+
  */
 
-#ifndef __TQMTEST_H
-#define __TQMTEST_H
+#ifndef __TQMA8QX_H
+#define __TQMA8QX_H
 
 #include <linux/sizes.h>
 #include <asm/arch/imx-regs.h>
