From 54b78919aa261cd8ca59a372cac9d6623b2a24b8 Mon Sep 17 00:00:00 2001
From: Markus Niebel <Markus.Niebel@tq-group.com>
Date: Thu, 17 Jan 2019 09:52:50 +0100
Subject: [PATCH 06/37] net: Kconfig: mark FEC_MXC compatible for MX7 / MX8MQ /
 MX8QXP

Signed-off-by: Markus Niebel <Markus.Niebel@tq-group.com>
---
 drivers/net/Kconfig | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/Kconfig b/drivers/net/Kconfig
index 6d9c51505e..880779ad7c 100644
--- a/drivers/net/Kconfig
+++ b/drivers/net/Kconfig
@@ -118,9 +118,9 @@ config ETHOC
 
 config FEC_MXC
 	bool "FEC Ethernet controller"
-	depends on MX5 || MX6
+	depends on MX5 || MX6 || MX7 || IMX8M || IMX8QXP
 	help
-	  This driver supports the 10/100 Fast Ethernet controller for
+	  This driver supports the 10/100/1000 Fast Ethernet controller for
 	  NXP i.MX processors.
 
 config FEC_MXC_SHARE_MDIO
-- 
2.17.1

