From afc6d4926842269ca392b2e81612192089ca72b1 Mon Sep 17 00:00:00 2001
From: Markus Niebel <Markus.Niebel@tq-group.com>
Date: Mon, 14 Jan 2019 12:12:35 +0100
Subject: [PATCH 08/24] tqma8mx: Kconfig: prepare bareboard support

Signed-off-by: Markus Niebel <Markus.Niebel@tq-group.com>
---
 board/tqc/tqma8mx/Kconfig | 42 +++++++++++++++++++++++++++++++++++++++
 1 file changed, 42 insertions(+)

diff --git a/board/tqc/tqma8mx/Kconfig b/board/tqc/tqma8mx/Kconfig
index 9095beb0ee..8348b011ee 100644
--- a/board/tqc/tqma8mx/Kconfig
+++ b/board/tqc/tqma8mx/Kconfig
@@ -9,6 +9,48 @@ config SYS_VENDOR
 config SYS_CONFIG_NAME
 	default "tqma8mx"
 
+choice
+	prompt "TQMa8MX CPU variant"
+	default TQMA8MX_MX8MQ
+	help
+	  Select cpu variant for TQMa8MX
+
+config TQMA8MX_MX8MQ
+	bool "TQMa8MX with i.MX8MQ"
+	help
+	  Select the CPU variant
+
+endchoice
+
+choice
+	prompt "TQMa8MX base board variant"
+	default MBA8MX
+	help
+	  Select base board for TQMa8MX
+
+config MBA8MX
+	bool "TQMa8MX on MBa8MX Starterkit"
+	select TQC_SDMMC
+	help
+	  Select the MBa8MX starterkit. This features a GigE Phy, USB, SD-Card
+	  etc.
+
+endchoice
+
+choice
+	prompt "TQMa8MX boot device"
+	default TQMBA8MX_SDCARD
+	help
+	  Select boot device for TQMa8MX
+
+config TQMBA8MX_SDCARD
+	bool "TQMa8MX from SD / e-MMC"
+	select SD_BOOT
+	help
+	  Select booting from SD / e-MMC
+
+endchoice
+
 source "board/tqc/common/Kconfig"
 
 endif
-- 
2.17.1

