From 82c27342e579712fae4a9a644af08534278263a2 Mon Sep 17 00:00:00 2001
From: Markus Niebel <Markus.Niebel@tq-group.com>
Date: Fri, 11 Jan 2019 09:40:27 +0100
Subject: [PATCH] board: tqma8mx: debug spl

Signed-off-by: Markus Niebel <Markus.Niebel@tq-group.com>
---
 board/tqc/tqma8mx/spl.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/board/tqc/tqma8mx/spl.c b/board/tqc/tqma8mx/spl.c
index d61592abe6..5ee96b34e6 100644
--- a/board/tqc/tqma8mx/spl.c
+++ b/board/tqc/tqma8mx/spl.c
@@ -29,6 +29,7 @@ DECLARE_GLOBAL_DATA_PTR;
 
 void spl_dram_init(void)
 {
+	puts("spl_dram_init\n");
 	/* ddr init */
 	ddr_init();
 }
@@ -91,6 +92,8 @@ static struct fsl_esdhc_cfg usdhc1_cfg = {
 int board_mmc_init(bd_t *bis)
 {
 	int ret;
+
+	puts("board_mmc_init\n");
 	/*
 	 * According to the board_mmc_init() the following map is done:
 	 * (U-Boot device node)    (Physical Port)
@@ -159,6 +162,8 @@ int power_init_board(void)
 
 void spl_board_init(void)
 {
+	puts("spl_board_init\n");
+
 	/* TrustZone Address Space Controller */
 	enable_tzc380();
 
@@ -207,7 +212,7 @@ void board_init_f(ulong dummy)
 	tqc_bb_board_init_f(dummy);
 
 	preloader_console_init();
-
+	puts("TQMa8MQ SPL\n");
 	/* Clear the BSS. */
 	memset(__bss_start, 0, __bss_end - __bss_start);
 
