From 8ed00e22a0b8da8969f705c137c9336a2edd9718 Mon Sep 17 00:00:00 2001
From: Markus Niebel <Markus.Niebel@tq-group.com>
Date: Mon, 19 Nov 2018 15:47:39 +0100
Subject: [PATCH] tqma8qxs: Kconfig: select dm drivers needed for boot

Signed-off-by: Markus Niebel <Markus.Niebel@tq-group.com>
---
 arch/arm/cpu/armv8/imx8/Kconfig | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/arm/cpu/armv8/imx8/Kconfig b/arch/arm/cpu/armv8/imx8/Kconfig
index adb2b8a8cf..0a1d182c0e 100644
--- a/arch/arm/cpu/armv8/imx8/Kconfig
+++ b/arch/arm/cpu/armv8/imx8/Kconfig
@@ -78,6 +78,11 @@ config TARGET_TQMA8QXS
 	bool "Support TQMa8xXS board"
 	select BOARD_LATE_INIT
 	select IMX8QXP
+	select CONFIG_DM_GPIO
+	select CONFIG_DM_ETH
+	select CONFIG_DM_SERIAL
+	select CONFIG_FSL_LPUART
+	select CONFIG_DM_MMC
 
 endchoice
 
