From c4b3377b7803674eaf80f658a41b71aec8068b05 Mon Sep 17 00:00:00 2001
From: Markus Niebel <Markus.Niebel@tq-group.com>
Date: Wed, 24 Oct 2018 17:14:34 +0200
Subject: [PATCH] tqma8qx: enable FAT_WRITE

Signed-off-by: Markus Niebel <Markus.Niebel@tq-group.com>
---
 include/configs/tqma8qx.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/configs/tqma8qx.h b/include/configs/tqma8qx.h
index efff1213f1..8f7e0f911a 100644
--- a/include/configs/tqma8qx.h
+++ b/include/configs/tqma8qx.h
@@ -15,6 +15,8 @@
 #define CONFIG_BOARD_EARLY_INIT_F
 #define CONFIG_ARCH_MISC_INIT
 
+#define CONFIG_FAT_WRITE
+
 /* Flat Device Tree Definitions */
 #define CONFIG_OF_BOARD_SETUP
 
