From 0a8d75dbb826bd2955025075370cb2155de4467c Mon Sep 17 00:00:00 2001
From: Markus Niebel <Markus.Niebel@tq-group.com>
Date: Wed, 24 Oct 2018 17:49:00 +0200
Subject: [PATCH] tqma8qx: configure memtest

Signed-off-by: Markus Niebel <Markus.Niebel@tq-group.com>
---
 include/configs/tqma8qx.h | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/include/configs/tqma8qx.h b/include/configs/tqma8qx.h
index 958613e94b..4070718483 100644
--- a/include/configs/tqma8qx.h
+++ b/include/configs/tqma8qx.h
@@ -304,6 +304,12 @@
 #define BOOTAUX_RESERVED_MEM_BASE 0x88000000
 #define BOOTAUX_RESERVED_MEM_SIZE 0x08000000 /* Reserve from second 128MB */
 
+#define CONFIG_CMD_MEMTEST
+#if defined(CONFIG_CMD_MEMTEST)
+#define CONFIG_SYS_MEMTEST_START  (BOOTAUX_RESERVED_MEM_BASE + BOOTAUX_RESERVED_MEM_SIZE)
+#define CONFIG_SYS_MEMTEST_END    (CONFIG_SYS_MEMTEST_START + PHYS_SDRAM_1_SIZE / 2)
+#endif
+
 #if defined(CONFIG_MBA8QX)
 #include "tqma8qx-mba8qx.h"
 #else
