From 1b7fab21043dd53f5aa5ba0c34340d7a02f38486 Mon Sep 17 00:00:00 2001
From: Markus Niebel <Markus.Niebel@tq-group.com>
Date: Thu, 14 Feb 2019 15:37:10 +0100
Subject: [PATCH] tqc/common: add missing include for in tqc_sdmmc

Signed-off-by: Markus Niebel <Markus.Niebel@tq-group.com>
---
 board/tqc/common/tqc_sdmmc.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/board/tqc/common/tqc_sdmmc.c b/board/tqc/common/tqc_sdmmc.c
index 755a0a7d6d..1a3cb8b3e9 100644
--- a/board/tqc/common/tqc_sdmmc.c
+++ b/board/tqc/common/tqc_sdmmc.c
@@ -8,6 +8,7 @@
 #include <asm/arch/sys_proto.h>
 #include <linux/errno.h>
 #include <asm/io.h>
+#include <mmc.h>
 #include <stdbool.h>
 
 static int check_mmc_autodetect(void)
