From 5cc587d6b41fbc2e835b867be61812a7479d559c Mon Sep 17 00:00:00 2001
From: Markus Niebel <Markus.Niebel@tq-group.com>
Date: Thu, 4 Oct 2018 17:35:48 +0200
Subject: [PATCH] net: fec_mxc: parse also phy-handle

Signed-off-by: Markus Niebel <Markus.Niebel@tq-group.com>
---
 drivers/net/fec_mxc.c | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/drivers/net/fec_mxc.c b/drivers/net/fec_mxc.c
index 2debbe911b..6ab55820e5 100644
--- a/drivers/net/fec_mxc.c
+++ b/drivers/net/fec_mxc.c
@@ -1245,7 +1245,18 @@ static int fec_phy_init(struct fec_priv *priv, struct udevice *dev)
 	int mask = 0xffffffff;
 
 #ifdef CONFIG_PHYLIB
-	mask = 1 << CONFIG_FEC_MXC_PHYADDR;
+	int offset = 0;
+	int reg = 0;
+
+	offset = fdtdec_lookup_phandle(gd->fdt_blob, dev_of_offset(dev),
+				       "phy-handle");
+	if (offset > 0) {
+		reg = fdtdec_get_int(gd->fdt_blob, offset, "reg", 0);
+		mask = 1 << reg;
+	} else {
+		printf("phy-handle does not exist under fec %s\n", dev->name);
+		mask = 1 << CONFIG_FEC_MXC_PHYADDR;
+	}
 #endif
 
 	phydev = phy_find_by_mask(priv->bus, mask, priv->interface);
