From e5b30c9a51147f9e8f7becd5ec64f3587ef93cf8 Mon Sep 17 00:00:00 2001
From: Markus Niebel <Markus.Niebel@tq-group.com>
Date: Fri, 28 Sep 2018 15:46:25 +0200
Subject: [PATCH] WIP: adapt for TQMaxX board part 6 - fix RAM layout

Signed-off-by: Markus Niebel <Markus.Niebel@tq-group.com>
---
 include/configs/tqma8qx.h | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/include/configs/tqma8qx.h b/include/configs/tqma8qx.h
index a6c5d83421..439c59d3b0 100644
--- a/include/configs/tqma8qx.h
+++ b/include/configs/tqma8qx.h
@@ -233,12 +233,12 @@
 #define CONFIG_SYS_MALLOC_LEN		((CONFIG_ENV_SIZE + (32*1024)) * 1024)
 
 #define CONFIG_SYS_SDRAM_BASE		0x80000000
-#define CONFIG_NR_DRAM_BANKS		3
+#define CONFIG_NR_DRAM_BANKS		1
 #define PHYS_SDRAM_1			0x80000000
-#define PHYS_SDRAM_2			0x880000000
 #define PHYS_SDRAM_1_SIZE		0x80000000	/* 2 GB */
-/* LPDDR4 board total DDR is 3GB */
-#define PHYS_SDRAM_2_SIZE		0x40000000	/* 1 GB */
+/* needed for loop in CPU code */
+#define PHYS_SDRAM_2			0x800000000
+#define PHYS_SDRAM_2_SIZE		0x0000000	/* not placed */
 
 /* Serial */
 #define CONFIG_BAUDRATE			115200
