From f23cce57fc0d3d2717634c562f052a5a64404cc6 Mon Sep 17 00:00:00 2001
From: Markus Niebel <Markus.Niebel@tq-group.com>
Date: Fri, 16 Nov 2018 10:41:24 +0100
Subject: [PATCH] tqma8qx: default DT to B0 Stepping

Signed-off-by: Markus Niebel <Markus.Niebel@tq-group.com>
---
 include/configs/tqma8qx.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/configs/tqma8qx.h b/include/configs/tqma8qx.h
index cf5c12977d..b6fff3fc97 100644
--- a/include/configs/tqma8qx.h
+++ b/include/configs/tqma8qx.h
@@ -132,7 +132,7 @@
 	"cntr_addr=0x88000000\0"			\
 	"cntr_file=os_cntr_signed.bin\0" \
 	"boot_fdt=try\0" \
-	"fdt_file=fsl-imx8qxp-tqma8qx-a0-mba8qx.dtb\0" \
+	"fdt_file=fsl-imx8qxp-tqma8qx-mba8qx.dtb\0" \
 	"mmcdev="__stringify(CONFIG_SYS_MMC_ENV_DEV)"\0" \
 	"mmcpart=" __stringify(CONFIG_SYS_MMC_IMG_LOAD_PART) "\0" \
 	"mmcroot=" CONFIG_MMCROOT " rootwait rw\0" \
