From 74a6e596edda8ec4e3b95bc5ede91e6f13334f51 Mon Sep 17 00:00:00 2001
From: Ye Li <ye.li@nxp.com>
Date: Mon, 10 Sep 2018 20:24:22 -0700
Subject: [PATCH] board: tqma8mx: select TQC_EEPROM

Signed-off-by: Markus Niebel <Markus.Niebel@tq-group.com>
---
 arch/arm/mach-imx/imx8m/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/mach-imx/imx8m/Kconfig b/arch/arm/mach-imx/imx8m/Kconfig
index 19f9297303..23399455c6 100644
--- a/arch/arm/mach-imx/imx8m/Kconfig
+++ b/arch/arm/mach-imx/imx8m/Kconfig
@@ -73,6 +73,7 @@ config TARGET_TQMA8MX
 	select MISC
 	select SUPPORT_SPL
 	select SPL
+	select TQC_EEPROM
 
 endchoice
 
