From ec19fe9afdd58d1bf16cff056ebbce163fe15bad Mon Sep 17 00:00:00 2001
From: Markus Niebel <Markus.Niebel@tq-group.com>
Date: Wed, 30 Jan 2019 16:21:06 +0100
Subject: [PATCH] board: tqma8mq: configure alternative mem test

Signed-off-by: Markus Niebel <Markus.Niebel@tq-group.com>
---
 include/configs/tqma8mx.h | 13 +++++++++++--
 1 file changed, 11 insertions(+), 2 deletions(-)

diff --git a/include/configs/tqma8mx.h b/include/configs/tqma8mx.h
index 88b683b598..a896f85cde 100644
--- a/include/configs/tqma8mx.h
+++ b/include/configs/tqma8mx.h
@@ -259,9 +259,18 @@
 
 #define CONFIG_NR_DRAM_BANKS		1
 
+#if defined(CONFIG_CMD_MEMTEST)
+/*
+ * Use alternative / extended memtest,
+ * leave 128 MiB free at start
+ * U-Boot is loaded to 0x40200000 (offset 2 MiB)
+ * and relocated at end of configured RAM
+ */
+#define CONFIG_SYS_ALT_MEMTEST
 #define CONFIG_SYS_MEMTEST_START	(CONFIG_SYS_SDRAM_BASE + SZ_128M)
-#define CONFIG_SYS_MEMTEST_END		(CONFIG_SYS_MEMTEST_START + SZ_512M)
-
+#define CONFIG_SYS_MEMTEST_END		(CONFIG_SYS_MEMTEST_START + (PHYS_SDRAM_SIZE / 4) * 3)
+#define CONFIG_SYS_MEMTEST_SCRATCH CONFIG_SYS_MEMTEST_END
+#endif
 
 #define CONFIG_BAUDRATE			115200
 
