From e43a9e12ca52dfe6d0374b1c5811817988f1ad87 Mon Sep 17 00:00:00 2001
From: Markus Niebel <Markus.Niebel@tq-group.com>
Date: Mon, 10 Sep 2018 11:12:15 +0200
Subject: [PATCH 023/188] spi: spidev: [HACK] add generic compatible for
 tq,testdev

Signed-off-by: Markus Niebel <Markus.Niebel@tq-group.com>
---
 drivers/spi/spidev.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/spi/spidev.c b/drivers/spi/spidev.c
index cda10719d1d1..092f8ae943f3 100644
--- a/drivers/spi/spidev.c
+++ b/drivers/spi/spidev.c
@@ -669,6 +669,7 @@ static const struct of_device_id spidev_dt_ids[] = {
 	{ .compatible = "lineartechnology,ltc2488" },
 	{ .compatible = "ge,achc" },
 	{ .compatible = "semtech,sx1301" },
+	{ .compatible = "tq,testdev" },
 	{},
 };
 MODULE_DEVICE_TABLE(of, spidev_dt_ids);
