From f39f96ec9a75d144cb72528507c58f385dd99951 Mon Sep 17 00:00:00 2001
From: Max Merchel <Max.Merchel@tq-group.com>
Date: Thu, 31 Jan 2019 08:40:19 +0100
Subject: [PATCH 088/188] pci: imx: handle phy ref clk source for i.MX7

Signed-off-by: Max Merchel <Max.Merchel@tq-group.com>
---
 drivers/pci/dwc/pci-imx6.c | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/drivers/pci/dwc/pci-imx6.c b/drivers/pci/dwc/pci-imx6.c
index 54459b52f526..a7cf3091290c 100644
--- a/drivers/pci/dwc/pci-imx6.c
+++ b/drivers/pci/dwc/pci-imx6.c
@@ -1303,8 +1303,16 @@ static void imx_pcie_init_phy(struct imx_pcie *imx_pcie)
 				"failed to enable pcie regulator\n");
 
 		/* pcie phy ref clock select; 1? internal pll : external osc */
-		regmap_update_bits(imx_pcie->iomuxc_gpr, IOMUXC_GPR12,
-				   IMX7D_GPR12_PCIE_PHY_REFCLK_SEL, 0);
+		if (imx_pcie->ext_osc) { /* external oscillator */
+			regmap_update_bits(imx_pcie->iomuxc_gpr,
+				IOMUXC_GPR12,
+				IMX7D_GPR12_PCIE_PHY_REFCLK_SEL, 0);
+		} else { /* internel oscillator */
+			regmap_update_bits(imx_pcie->iomuxc_gpr,
+				IOMUXC_GPR12,
+				IMX7D_GPR12_PCIE_PHY_REFCLK_SEL,
+				IMX7D_GPR12_PCIE_PHY_REFCLK_SEL);
+		}
 	} else if (imx_pcie->variant == IMX6SX) {
 		/* Force PCIe PHY reset */
 		regmap_update_bits(imx_pcie->iomuxc_gpr, IOMUXC_GPR5,
