From a801387767e4883e1c657a73949cc2fc3dcecbef Mon Sep 17 00:00:00 2001
From: Markus Niebel <Markus.Niebel@tq-group.com>
Date: Tue, 19 Feb 2019 17:22:20 +0100
Subject: [PATCH 136/188] net: dp83867: make ledcfg register config real
 optional

Signed-off-by: Markus Niebel <Markus.Niebel@tq-group.com>
---
 drivers/net/phy/dp83867.c | 12 +++++++++---
 1 file changed, 9 insertions(+), 3 deletions(-)

diff --git a/drivers/net/phy/dp83867.c b/drivers/net/phy/dp83867.c
index 2f32fedadbf6..c523fb14ce9f 100644
--- a/drivers/net/phy/dp83867.c
+++ b/drivers/net/phy/dp83867.c
@@ -196,18 +196,24 @@ static int dp83867_of_init(struct phy_device *phydev)
 
 	ret = of_property_read_u32(of_node, "ti,led-cfg1",
 				&dp83867->led_cfg1);
-	if (ret)
+	if (ret) {
 		dev_dbg(dev, "could not find ti,led-cfg1\n");
+		dp83867->led_cfg2 = 0x6150;
+	}
 
 	ret = of_property_read_u32(of_node, "ti,led-cfg2",
 				&dp83867->led_cfg2);
-	if (ret)
+	if (ret) {
 		dev_dbg(dev, "could not find ti,led-cfg2\n");
+		dp83867->led_cfg2 = 0x2222;
+	}
 
 	ret = of_property_read_u32(of_node, "ti,led-cfg3",
 				&dp83867->led_cfg3);
-	if (ret)
+	if (ret) {
 		dev_dbg(dev, "could not find ti,led-cfg3\n");
+		dp83867->led_cfg3 = 0x02;
+	}
 
 	ret = of_property_read_u32(of_node, "ti,fifo-depth",
 				   &dp83867->fifo_depth);
