diff --git a/ppa/build b/ppa/build
index 3c5eb66..bb006c5 100755
--- a/ppa/build
+++ b/ppa/build
@@ -539,7 +539,7 @@ until [ "$1" = "" ]; do
             trailer=":$persona"
         fi
         echo $action soc-$device$trailer
-        make -j24 $target personality=$persona >> ../bldlog.txt
+        make -j1 $target personality=$persona >> ../bldlog.txt
 
     else
         action=building
@@ -649,7 +649,7 @@ until [ "$1" = "" ]; do
 
         if [ "$tools_built" == "false" ]; then
          cd ./tools
-         make -j24  >> ../bldlog.txt
+         make -j1  >> ../bldlog.txt
          cd - >> ../bldlog.txt
             tools_built="true"
         fi
@@ -668,7 +668,7 @@ until [ "$1" = "" ]; do
         echo >> ../bldlog.txt
         echo $action $bld_mode $target [$make_opts $test_opt ] $device$trailer >> ../bldlog.txt
 
-        make -j24 $target $make_opts $test_opt personality=$persona >> ../bldlog.txt
+        make -j1 $target $make_opts $test_opt personality=$persona >> ../bldlog.txt
     fi
     cd ..
 
