From 4dcb7a6fc37fd7d73f4df4f0a579c6015e01d352 Mon Sep 17 00:00:00 2001
From: Markus Niebel <Markus.Niebel@tq-group.com>
Date: Tue, 23 Oct 2018 12:07:09 +0200
Subject: [PATCH] arm64: platforms: select CLKSRC_MMIO and RESET_CONTROLLER for
 i.MX8

Signed-off-by: Markus Niebel <Markus.Niebel@tq-group.com>
---
 arch/arm64/Kconfig.platforms | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/arch/arm64/Kconfig.platforms b/arch/arm64/Kconfig.platforms
index ef3f5cf9ac23..69219e5b7dbf 100644
--- a/arch/arm64/Kconfig.platforms
+++ b/arch/arm64/Kconfig.platforms
@@ -232,6 +232,8 @@ config ARCH_ZYNQMP
 
 config ARCH_FSL_IMX8QM
 	bool "Freescale i.MX8QM SOC"
+	select CLKSRC_MMIO
+	select RESET_CONTROLLER
 	select PINCTRL
 	select PINCTRL_IMX
 	select PINCTRL_IMX8QM
@@ -242,6 +244,8 @@ config ARCH_FSL_IMX8QM
 
 config ARCH_FSL_IMX8QP
 	bool "Freescale i.MX8QP SOC"
+	select CLKSRC_MMIO
+	select RESET_CONTROLLER
 	select PINCTRL
 	select PINCTRL_IMX
 	select PINCTRL_IMX8QM
@@ -253,6 +257,8 @@ config ARCH_FSL_IMX8QP
 
 config ARCH_FSL_IMX8QXP
 	bool "Freescale i.MX8QXP SOC"
+	select CLKSRC_MMIO
+	select RESET_CONTROLLER
 	select PINCTRL
 	select PINCTRL_IMX
 	select PINCTRL_IMX8QXP
@@ -265,6 +271,8 @@ config ARCH_FSL_IMX8MQ
 	bool "Freescale i.MX8MQ SOC"
 	select ARCH_MXC
 	select SERIAL_IMX
+	select CLKSRC_MMIO
+	select RESET_CONTROLLER
 	select PINCTRL
 	select PINCTRL_IMX
 	select PINCTRL_IMX8MQ
@@ -278,6 +286,8 @@ config ARCH_FSL_IMX8MM
 	bool "Freescale i.MX8MM SOC"
 	select ARCH_MXC
 	select SERIAL_IMX
+	select CLKSRC_MMIO
+	select RESET_CONTROLLER
 	select PINCTRL
 	select PINCTRL_IMX
 	select PINCTRL_IMX8MM
-- 
2.17.1

