diff --git a/plat/renesas/rcar/bl2_rcar_setup.c b/plat/renesas/rcar/bl2_rcar_setup.c
index c442573..ff1e797 100644
--- a/plat/renesas/rcar/bl2_rcar_setup.c
+++ b/plat/renesas/rcar/bl2_rcar_setup.c
@@ -887,7 +887,9 @@ void bl2_plat_flush_bl31_params(void)
 	uint32_t val;
 	uint32_t modemr;
 	uint32_t modemr_boot_dev;
+#if RCAR_SECURE_BOOT
 	uint32_t lcs;
+#endif
 
 	modemr = mmio_read_32(RCAR_MODEMR);
 	modemr_boot_dev = modemr & MODEMR_BOOT_DEV_MASK;
@@ -958,6 +960,7 @@ void bl2_plat_flush_bl31_params(void)
 	mmio_write_32(IPMMUMM_IMSCTLR, IPMMUMM_IMSCTLR_ENABLE);
 	mmio_write_32(IPMMUMM_IMAUXCTLR, IPMMUMM_IMAUXCTLR_NMERGE40_BIT);
 
+#if RCAR_SECURE_BOOT
 	val = ROM_GetLcs(&lcs);
 	if (val != 0U) {
 		ERROR("BL2: Failed to get the LCS. (%d)\n", val);
@@ -969,6 +972,7 @@ void bl2_plat_flush_bl31_params(void)
 			mmio_read_32(P_ARMREG_SEC_CTRL) & 
 				~((uint32_t)P_ARMREG_SEC_CTRL_PROT));
 	}
+#endif
 
 	/* disable the System WDT, FIQ and GIC	*/
 	bl2_swdt_release();
