diff --git a/plat/renesas/rcar/drivers/board/board.c b/plat/renesas/rcar/drivers/board/board.c
index 225ff93..1414d09 100644
--- a/plat/renesas/rcar/drivers/board/board.c
+++ b/plat/renesas/rcar/drivers/board/board.c
@@ -6,7 +6,9 @@
 
 #include <stdint.h>
 #include <iic_dvfs.h>
+#include <mmio.h>
 #include "board.h"
+#include "../../rcar_def.h"
 
 
 /************************************************************************
@@ -39,6 +41,10 @@
 
 #define BOARD_ID_UNKNOWN	(0xFFU)
 
+#define	GPIO_INDT5	0xE605500C
+#define	GP5_19_BIT	(0x01 << 19)
+#define	GP5_21_BIT	(0x01 << 21)
+#define	GP5_25_BIT	(0x01 << 25)
 
 /************************************************************************
  * Global variables
@@ -62,6 +68,10 @@ int32_t get_board_type(uint32_t *type, uint32_t *rev)
 {
 	int32_t ret = 0;
 	uint8_t read_rev;
+#if	(RZG_HIHOPE_RZG2N) | (RZG_HIHOPE_RZG2M)
+	uint32_t reg, boardInfo;
+#endif	/* RZG_HIHOPE_RZG2N | RZG_HIHOPE_RZG2M */
+
 	static uint8_t g_board_id = BOARD_ID_UNKNOWN;
 	const uint8_t board_tbl[][8U] = {
 		[BOARD_SALVATOR_X]	= {0x10U, 0x11U, 0xFFU, 0xFFU,
@@ -82,7 +92,7 @@ int32_t get_board_type(uint32_t *type, uint32_t *rev)
 					   0xFFU, 0xFFU, 0xFFU, 0xFFU},
 		[BOARD_HIHOPE_RZG2M]	= {0x10U, 0xFFU, 0xFFU, 0xFFU,
 					   0xFFU, 0xFFU, 0xFFU, 0xFFU},
-		[BOARD_HIHOPE_RZG2N]	= {0x10U, 0xFFU, 0xFFU, 0xFFU,
+		[BOARD_HIHOPE_RZG2N]	= {0x20U, 0xFFU, 0xFFU, 0xFFU,
 					   0xFFU, 0xFFU, 0xFFU, 0xFFU},
 	};
 
@@ -106,8 +116,34 @@ int32_t get_board_type(uint32_t *type, uint32_t *rev)
 
 	*type = ((uint32_t)g_board_id & BOARD_CODE_MASK) >> BOARD_CODE_SHIFT;
 	if (*type < (sizeof(board_tbl) / sizeof(board_tbl[0]))) {
+#if (RZG_HIHOPE_RZG2M)
+		reg = mmio_read_32(RCAR_PRR);
+		if (RCAR_M3_CUT_VER11 == (reg & RCAR_CUT_MASK))
+		{
+			read_rev = (uint8_t)(g_board_id & BOARD_REV_MASK);
+			*rev = board_tbl[*type][read_rev];
+		}
+		else
+		{
+			boardInfo = mmio_read_32(GPIO_INDT5) & (GP5_19_BIT |GP5_21_BIT);
+			*rev = (((boardInfo & GP5_19_BIT) >> 14) | ((boardInfo & GP5_21_BIT) >> 17)) + 0x30;
+		}
+#elif (RZG_HIHOPE_RZG2N)
+		reg = mmio_read_32(GPIO_INDT5);
+		if (reg & GP5_25_BIT)
+		{
+			read_rev = (uint8_t)(g_board_id & BOARD_REV_MASK);
+			*rev = board_tbl[*type][read_rev];
+		}
+		else
+		{
+			boardInfo = reg & (GP5_19_BIT |GP5_21_BIT);
+			*rev = (((boardInfo & GP5_19_BIT) >> 14) | ((boardInfo & GP5_21_BIT) >> 17)) + 0x30;
+		}
+#else
 		read_rev = (uint8_t)(g_board_id & BOARD_REV_MASK);
 		*rev = board_tbl[*type][read_rev];
+#endif
 	} else {
 		/* If there is no revision information, set Rev0.0. */
 		*rev = 0x00U;

diff --git a/plat/renesas/rcar/ddr/ddr_b/boot_init_dram_config.c b/plat/renesas/rcar/ddr/ddr_b/boot_init_dram_config.c
index 5d13e0d..2dfa187 100644
--- a/plat/renesas/rcar/ddr/ddr_b/boot_init_dram_config.c
+++ b/plat/renesas/rcar/ddr/ddr_b/boot_init_dram_config.c
@@ -16,18 +16,29 @@
  *	PLEASE SET board number or board judge function
  ******************************************************************************/
 #if (RZG_HIHOPE_RZG2M) // RZG_HIHOPE_RZG2M
+#define	GPIO_INDT5	0xE605500C
+#define	LPDDR4_2RANK	(0x01 << 25)
+
 static uint32_t boardcnf_get_brd_type(void){
 	uint32_t reg;
+	uint32_t boardInfo;
 
 	reg = mmio_read_32(RCAR_PRR);
-
 	if (RCAR_M3_CUT_VER11 == (reg & RCAR_CUT_MASK))
 	{
 		return (23);
 	}
 	else
 	{
-		return (22);
+		boardInfo = mmio_read_32(GPIO_INDT5);
+		if (boardInfo & LPDDR4_2RANK)
+		{
+			return (23);
+		}
+		else
+		{
+			return (22);
+		}
 	}
 }
 #elif (RZG_HIHOPE_RZG2N) // RZG_HIHOPE_RZG2N

diff --git a/plat/renesas/rcar/pfc/M3/pfc_init_m3.c b/plat/renesas/rcar/pfc/M3/pfc_init_m3.c
index e53467a..3b47fc8 100644
--- a/plat/renesas/rcar/pfc/M3/pfc_init_m3.c
+++ b/plat/renesas/rcar/pfc/M3/pfc_init_m3.c
@@ -1353,8 +1353,7 @@ void pfc_init_m3(void)
 			       | GPSR2_IRQ2
 			       | GPSR2_IRQ1
 			       | GPSR2_IRQ0);
-	pfc_reg_write(PFC_GPSR3, GPSR3_SD0_WP
-			       | GPSR3_SD0_CD
+	pfc_reg_write(PFC_GPSR3, GPSR3_SD0_CD
 			       | GPSR3_SD1_DAT3
 			       | GPSR3_SD1_DAT2
 			       | GPSR3_SD1_DAT1

diff --git a/plat/renesas/rcar/pfc/M3N/pfc_init_m3n.c b/plat/renesas/rcar/pfc/M3N/pfc_init_m3n.c
index 3495d95..fc1cdb2 100644
--- a/plat/renesas/rcar/pfc/M3N/pfc_init_m3n.c
+++ b/plat/renesas/rcar/pfc/M3N/pfc_init_m3n.c
@@ -1260,8 +1260,7 @@ void pfc_init_m3n(void)
 			       | GPSR2_IRQ2
 			       | GPSR2_IRQ1
 			       | GPSR2_IRQ0);
-	pfc_reg_write(PFC_GPSR3, GPSR3_SD0_WP
-			       | GPSR3_SD0_CD
+	pfc_reg_write(PFC_GPSR3, GPSR3_SD0_CD
 			       | GPSR3_SD1_DAT3
 			       | GPSR3_SD1_DAT2
 			       | GPSR3_SD1_DAT1
